src/elm/Actor/Component/AiComponent/AdventAiComponent.elm

```diff
@@ -0,0 +1,343 @@
+    Dict.insert
+        "transform"
+        (Actor.TransformComponent
+            { position = newPosition
+            , movingState = Actor.NotMoving
+            }
+        )
+        actor.components
+        |> Common.updateComponents actor
+        |> Common.updateActor level.actors
+        |> Common.updateActors level
+        |> Common.removeActorFromIndexByPosition oldPosition actor.id
+        |> Common.addActorToIndex newPosition actor.id
+
+
+setupSearch : Position -> TargetSearch
+setupSearch position =
+    { foundPositions = Dict.fromList [ ( ( position.x, position.y ), () ) ]
+    , paths =
+        [ { directions = []
+          , currentPosition = position
+          }
+        ]
+    }
+
+
+search : Level -> String -> TargetSearch -> List Action
+search level enemyTag targetSearch =
+    List.range 0 30
+        |> List.foldl
+            (\step acc -> move level enemyTag step acc)
+            (Continue targetSearch)
+        |> asActions
+
+
+asActions : SearchAcc -> List Action
+asActions acc =
+    case acc of
+        Stop actions ->
+            actions
+
+        Continue _ ->
+            []
+
+
+move : Level -> String -> Int -> SearchAcc -> SearchAcc
+move level enemyTag step acc =
+    case acc of
+        Stop searchResult ->
+            Stop searchResult
+
+        Continue targetSearch ->
+            let
+                newTargetSearch =
+                    walkOneStep targetSearch
+            in
+            case ( step, getEnemiesOnCurrentPositions level newTargetSearch enemyTag ) of
+                ( _, [] ) ->
+                    newTargetSearch.paths
+                        |> List.filter
+                            (\path ->
+                                Common.isEmpty path.currentPosition level
+                            )
+                        |> (\newPaths ->
+                                Continue { newTargetSearch | paths = newPaths }
+                           )
+
+                -- Enemy is near, we should fight!
+                -- Sort on health
+                ( 0, enemies ) ->
+                    enemies
+                        |> List.sortBy
+                            (\enemy ->
+                                ( enemy.healthData.health, enemy.path.currentPosition.y, enemy.path.currentPosition.x )
+                            )
+                        |> List.head
+                        |> Maybe.map (\enemy -> Stop [ Attack enemy.actor ])
+                        -- Should not happen
+                        |> Maybe.withDefault (Continue newTargetSearch)
+
+                -- Enemy is near, we should walk and then fight!
+                -- Sort on health
+                ( 1, enemies ) ->
+                    enemies
+                        |> List.sortBy
+                            (\enemy ->
+                                ( enemy.healthData.health, enemy.path.currentPosition.y, enemy.path.currentPosition.x )
```

Comment:
This might be wrong. We need to first sort on position (that determines where to move to)
And then we need to filter enemies reachable from that position. And sort those on health.